src/components/MainModeIcon.jsx

```diff
@@ -1,55 +1,15 @@
 import React, { useMemo } from 'react'
 import PropTypes from 'prop-types'
 
-import Icon from 'cozy-ui/transpiled/react/Icon'
+import Avatar from 'cozy-ui/transpiled/react/Avatar'
 
-import BikeIcon from 'src/assets/icons/avatar-bike.svg'
-import BusIcon from 'src/assets/icons/avatar-bus.svg'
-import CarIcon from 'src/assets/icons/avatar-car.svg'
-import PlaneIcone from 'src/assets/icons/avatar-plane.svg'
-import SubwayIcon from 'src/assets/icons/avatar-subway.svg'
-import TrainIcon from 'src/assets/icons/avatar-train.svg'
-import WalkIcon from 'src/assets/icons/avatar-walk.svg'
-import UnknownIcon from 'src/assets/icons/icon-question-mark.svg'
-
-import {
-  AIR_MODE,
-  BICYCLING_MODE,
-  BUS_MODE,
-  CAR_MODE,
-  SUBWAY_MODE,
-  TRAIN_MODE,
-  WALKING_MODE,
-  UNKNOWN_MODE
-} from 'src/constants/const'
+import { pickModeIcon } from 'src/components/helpers'
 import { getMainMode } from 'src/lib/trips'
 
 const MainModeIcon = ({ trip }) => {
```

Comment:
this complete component should be memoized ;)

Reply:
I see that almost all the components are memoized now: what are the rules to memo or not a component?

Reply:
On mémoize tout depuis un moment par défaut, ça fait plus de bien que de mal même si mal employé. En théorie on évite quand même lorsque qu'il n'y a pas de props (puisque ça ne sert à rien dans ce cas)